project3.cc

```diff
@@ -0,0 +1,599 @@
+//TODO add stuff here
+
+#include "project3.h"
+#include <iomanip>
+using namespace std;
+
+/*
+ * Methods of class ThreadedBinarySearchTree
+ *
+ */
+
+typedef ThreadedBinarySearchTree::Iterator   Iterator;
+
+ThreadedBinarySearchTree::ThreadedBinarySearchTree()
+  : _header(new  Node())
+{
+  _header->_lchild = makeThread(_header);
+  _header->_rchild = _header;
+}
+
+/*
+* Destructor called when the user clicks q in the program
+* Iterators through the program deleting all nodes
+* To ensure we are a good steward of nodes
+*/
+ThreadedBinarySearchTree::~ThreadedBinarySearchTree()
+{
+  //Adam and Dr Tuck
+  Iterator it;
+  Node *tempnode;
+
+  for (it = this->inorder_begin(); it != this->end(); it++)
+  {
+    //temp node = to pointer
+    tempnode = it._ptr;
+    delete it._ptr;
+  }
+  // delete pointer;
+
+  delete _header;
+}
+
+
+/*
+ * Insert function called when the user calls i in the program
+ * PARAM-key: the string used to determine tree position
+ * PARAM-value: value that for the node
+ */
+void ThreadedBinarySearchTree::insert(string key, int value)
+{
+  //Deal with the case that are first item is the header
+  if (empty())
+  {
+    Node *newnode = new Node();
+    _header->_lchild = newnode;
+
+    newnode->_lchild = makeThread(_header);
+    newnode->_rchild = makeThread(_header);
+    newnode->_value = value;
+    newnode->_key = key;
+  }
+  else
+  {
+    ThreadedBinarySearchTree::insertr(key, value, _header->_lchild);
+  }
+}
+
+
+/*
+ * Recursive insertion called by insert()
+ * PARAM-key: the string used to determine tree position
```

Comment:
"PARAM-key" is interesting syntax.  Did you find it or invent it?
I like listing each parameter and explaining it.  That's a great habit.  
I'd prefer not to have the dash, which makes "PARAM-key" look like it's all one word.  
"PARAM key: the string..." or just "  key: the string..." seems clearer to me.